### PR TITLE
[SYCL] Get sampler and accessor types without using named fields of kernel object

### PR DESCRIPTION
**Pull request by schittir**
Use the types of the parameters of __init method instead

Signed-off-by: Sindhu Chittireddy <sindhu.chittireddy@intel.com>

### CODE REVIEW

File: clang/test/SemaSYCL/wrapped-accessor.cpp [hunk 1]
```diff
@@ -52,8 +52,8 @@ int main() {
 // CHECK-NEXT: ImplicitCastExpr {{.*}} '__global int *' lvalue <NoOp>
 // CHECK-NEXT: DeclRefExpr {{.*}} '__global int *' lvalue ParmVar {{.*}} '_arg_accessor' '__global int *'
 // CHECK-NEXT: ImplicitCastExpr {{.*}} 'range<1>':'cl::sycl::range<1>' <NoOp>
-// CHECK-NEXT: DeclRefExpr {{.*}} 'range<1>':'cl::sycl::range<1>' lvalue ParmVar {{.*}} '_arg_AccessRange' 'range<1>':'cl::sycl::range<1>'
+// CHECK-NEXT: DeclRefExpr {{.*}} 'range<1>':'cl::sycl::range<1>' lvalue ParmVar {{.*}} '{{[a-zA-Z_]+}}' 'range<1>':'cl::sycl::range<1>'
```
Comment (bader) [inline clang/test/SemaSYCL/wrapped-accessor.cpp]: I think it's better to define variables for kernel parameters and validate that correct parameter is used, if the intention is to omit "exact name" check.

File: clang/test/SemaSYCL/wrapped-accessor.cpp [hunk 2]
```diff
@@ -25,9 +25,9 @@ int main() {
 // Check parameters of the kernel
 // CHECK: ParmVarDecl {{.*}} used _arg_ 'AccWrapper<cl::sycl::accessor<int, 1, cl::sycl::access::mode::read_write, cl::sycl::access::target::global_buffer, cl::sycl::access::placeholder::false_t> >':'AccWrapper<cl::sycl::accessor<int, 1, cl::sycl::access::mode::read_write, cl::sycl::access::target::global_buffer, cl::sycl::access::placeholder::false_t> >'
 // CHECK: ParmVarDecl {{.*}} used _arg_accessor '__global int *'
-// CHECK: ParmVarDecl {{.*}} used _arg_AccessRange 'range<1>':'cl::sycl::range<1>'
-// CHECK: ParmVarDecl {{.*}} used _arg_MemRange 'range<1>':'cl::sycl::range<1>'
-// CHECK: ParmVarDecl {{.*}} used _arg_Offset 'id<1>':'cl::sycl::id<1>'
+// CHECK: ParmVarDecl {{.*}} used {{[a-zA-Z_]+}} 'range<1>':'cl::sycl::range<1>'
```
Comment (bader) [inline clang/test/SemaSYCL/wrapped-accessor.cpp]: I think numbers are also should be allowed.

**Comment by bader** [comment]
Please, rebase your patch to the tip of the branch to avoid false positives in CI system.

File: clang/test/SemaSYCL/wrapped-accessor.cpp (hunk 1 above)
Comment (schittir) [reply clang/test/SemaSYCL/wrapped-accessor.cpp]: Done.

File: clang/test/SemaSYCL/wrapped-accessor.cpp (hunk 2 above)
Comment (schittir) [reply clang/test/SemaSYCL/wrapped-accessor.cpp]: Done.

**Pull request merged**